scripts/tablegen.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+
+# Copyright 2017,2018 Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import string
+import sys
+import linecache
+
+from ruamel.yaml import YAML
+
+# Reads a documented Helm values.yaml file and produces a
+# MD formatted table.  pip install ruamel to obtain the proper
+# YAML decoder.  ruamel maintains ordering and comments.  The
+# comments are needed in order to decode the commented helm
+# values.yaml file
+
+def endOfTheList(context, lineNum, lastLineNum, totalNum):
+    flag = 0
+    valueList = []
+    nextLineNum = lineNum + 1
+    currentLine = context[lastLineNum]
+    whitespaces = (len(currentLine) - len(currentLine.lstrip())) / 2
+
+    if lineNum != lastLineNum:
+        return False, valueList
+
+    for nextLineNum in range(lineNum + 1, totalNum):
+        nextLine = context[nextLineNum]
+        if  len(nextLine.lstrip()) != 0 and '#' != nextLine.lstrip()[0] and ':' in nextLine:
+            if whitespaces >= (len(nextLine) - len(nextLine.lstrip())) / 2:
+                if flag == 0:
+                    valueList.append(currentLine.split(':', 1)[1].strip())
+                    return True, valueList
```

Comment:
this is odd logic - why not just return True, valueList at the scope of the if whitespace conditional?

Reply:
Yes, we can use one `return True, valueList`

```
if flag == 0:
    valueList.append(currentLine.split(':', 1)[1].strip())

return True, valueList
```

Reply:
done